transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -9,75 +11,72 @@ import {
   SubscribeMessage,
   WebSocketGateway,
   WebSocketServer,
+  WsException,
 } from '@nestjs/websockets';
 import { Socket, Server } from 'socket.io';
-import { v4 as uuidv4 } from 'uuid';
+import { BadRequestTransformationFilter } from '../shared/filters/bad-request-transformation.filter';
 import { WsAuthenticatedGuard } from '../shared/guards/ws-authenticated.guard';
-import { User } from '../user/infrastructure/db/user.entity';
-
-type ChatroomMessage = {
-  id: string;
-  user: User;
-  content: string;
-  createdAt: number;
-  chatroomId: string;
-};
-
-type ChatroomId = string;
+import { ChatService } from './chat.service';
+import { ChatroomMemberService } from './chatroom/chatroom-member/chatroom-member.service';
+import { CreateChatroomMessageDto } from './chatroom/chatroom-message/dto/create-chatroom-message.dto';
 
 @WebSocketGateway({ path: '/api/v1/socket.io' })
 @UseGuards(WsAuthenticatedGuard)
 @UseInterceptors(ClassSerializerInterceptor)
+@UseFilters(BadRequestTransformationFilter)
 export class ChatGateway {
   @WebSocketServer() server!: Server;
-  chatrooms = new Map<ChatroomId, Set<ChatroomMessage>>();
+
+  constructor(
+    private chatService: ChatService,
+    private chatroomMemberService: ChatroomMemberService,
+  ) {}
 
   @SubscribeMessage('chatroomMessage')
-  handleMessage(
-    @MessageBody() data: { chatroomId: ChatroomId; content: string },
+  async handleMessage(
+    @MessageBody()
+    createChatroomMessageDto: CreateChatroomMessageDto,
     @ConnectedSocket() client: Socket,
   ) {
+    const { chatroomId, content } = createChatroomMessageDto;
     const user = client.request.user;
```

Comment:
why cant we use the @GetUser injectable for this?

Reply:
I think the `GetUser` decorator will not work in the WebSocket context, but we could create another one :+1:.
Thanks, I will give it a try.

Reply:
Update: I've tried to create a decorator for this and could not get it working in the WebSocket context :pensive:

This is what I tried:

<details>

```typescript
import { createParamDecorator, ExecutionContext } from '@nestjs/common';

export const UserWs = createParamDecorator(
  (property: string, ctx: ExecutionContext) => {
    const client = ctx.switchToWs().getClient();
    const user = client.request.user;

    return property ? user?.[property] : user;
  },
);
```

</details>